### PR TITLE
Awsconfigrules

### PR DESCRIPTION
**Pull request by SonalOjha**
Changes to introduce a new security group checker rule

### CODE REVIEW

File: lib/aws/ec2.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            return;
+        }
+
+        self.callbackFind = function(data) {
+            var hasRules = false;
+            if (data.SecurityGroups.length !== 0) {
+                for( var item in data.SecurityGroups) {
+                    var permissions = data.SecurityGroups[item].IpPermissions;
+                    console.log("Inbound Rules");
+                    for ( var it in permissions) {
```
Comment (AlexOugh) [inline lib/aws/ec2.js]: Any reason why loop? Isn't it simple to check the permissions.length and print out permissions as a whole?

File: lib/aws/ec2.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                        hasRules = true;
+                        console.log(data.SecurityGroups[item].IpPermissions[it]);
+                    }
+                    var permissionsEgress = data.SecurityGroups[item].IpPermissionsEgress;
+                    console.log("Outbound Rules");
+                    for ( var it2 in permissionsEgress) {
```
Comment (AlexOugh) [inline lib/aws/ec2.js]: same with inbound rules

File: lib/aws/ec2.js (hunk 1 above)
Comment (SonalOjha) [inline lib/aws/ec2.js]: Yes, we wanted to add whats rules were added in logs so looping through. Actually this changes in awsconfigrules/all_region_vpcs/MA-67 branch.

File: lib/aws/ec2.js (hunk 2 above)
Comment (SonalOjha) [inline lib/aws/ec2.js]: this changes too, see above comment

File: lib/aws/awsconfig.js [hunk 3]
```diff
@@ -20,6 +20,73 @@ function AWSConfig() {
     return configService;
   }
 
+  me.enableRule = function(input) {
+    var self = arguments.callee;
+    var configService = me.findService(input);
+    if(input.owner == "CUSTOM_LAMBDA"){
+    var params = {
```
Comment (AlexOugh) [inline lib/aws/awsconfig.js]: How about this?

`var params = {`

    ConfigRule:{
        ConfigRuleName:input.ruleName,
        Description:input.descript,
        Source: {
            Owner: input.owner,
            SourceIdentifier: input.sourceID,
        },
        Scope: {ComplianceResourceTypes: input.resourceType},
        InputParameters: input.params
    }    
`};`

`if(input.owner == "CUSTOM_LAMBDA"){`

  `params.ConfigRule.Source.SourceDetails = [`

      {
          "EventSource": "aws.config",
          "MessageType": "ConfigurationItemChangeNotification"
      }
  `]`

 `}`

File: lib/aws/ec2.js (hunk 1 above)
Comment (AlexOugh) [inline lib/aws/ec2.js]: but we can print 'permission' as a whole

File: lib/aws/awsconfig.js (hunk 3 above)
Comment (SonalOjha) [inline lib/aws/awsconfig.js]: This is a common code and isnt only meant when input.owner is CUSTOM_LAMBDA. The input has the relevant values.
The comment "**//Owner: "CUSTOM_LAMBDA"**," is not an inline comment and can be removed.

Comment (AlexOugh) [inline lib/aws/awsconfig.js]: I'm not sure what you mean. What I'm trying to point out is that the only difference is to set 'SourceDetails' inside ConfigRule.Source.

Comment (SonalOjha) [inline lib/aws/awsconfig.js]: ohhkk, got it

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 
-BUILD_SUBDIRS := lib billingalert awsconfig cloudtrail alarmalert volume
```
Comment (AlexOugh) [inline Makefile]: Can you get the latest of develop branch? It looks like the addition of 'volume' is missing in your branch

File: lib/aws/ec2.js (hunk 1 above)
Comment (SonalOjha) [inline lib/aws/ec2.js]: Agreed this should have been cleaned.

File: Makefile (hunk 4 above)
Comment (SonalOjha) [inline Makefile]: git merge doesnt show any changes, I have made an explicit change.

**Pull request merged**